src/task.py

```diff
@@ -0,0 +1,486 @@
+import argparse
+import itertools
+import json
+import math
+import multiprocessing
+import os
+import re
+import shutil
+import subprocess
+import tarfile
+import threading
+import uuid
+from concurrent.futures import ProcessPoolExecutor
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Any, Dict, List, Optional, TextIO, Tuple, Union
+
+import requests
+from google.cloud.storage import Client  # type: ignore
+from osgeo import gdal, gdalconst  # type: ignore
+from pyproj import Geod  # type: ignore
+from shapely import wkt as shp_wkt  # type: ignore
+from shapely.validation import explain_validity  # type: ignore
+from task_base import HIITask  # type: ignore
+
+import config
+import raster_utils
+from timer import Timer
+
+gdal.SetConfigOption("GDAL_CACHEMAX", "512")
+gdal.SetConfigOption("GDAL_SWATH_SIZE", "512")
+gdal.SetConfigOption("GDAL_MAX_DATASET_POOL_SIZE", "400")
+
+
+def run_in_thread(fn):
+    def run(*k, **kw):
+        t = threading.Thread(target=fn, args=k, kwargs=kw)
+        t.start()
+        return
+
+    return run
+
+
+def _rasterize(
+    in_file: Union[str, Path],
+    output_path: Union[str, Path],
+    output_bounds: Optional[List[float]] = None,
+) -> Path:
+    output_bounds = output_bounds or [-180.0, -90.0, 180.0, 90.0]
+    opts = gdal.RasterizeOptions(
+        format="GTiff",
+        outputType=gdalconst.GDT_Byte,
+        noData=0,
+        initValues=0,
+        burnValues=1,
+        xRes=0.003,
+        yRes=0.003,
+        targetAlignedPixels=True,
+        outputSRS="EPSG:4326",
+        outputBounds=output_bounds,
+        creationOptions=[
+            "TILED=YES",
+            "BLOCKXSIZE=1024",
+            "BLOCKYSIZE=1024",
+        ],
+    )
+    gdal.Rasterize(str(output_path), str(in_file), options=opts)
+
+    return Path(output_path)
+
+
+class ConversionException(Exception):
+    pass
+
+
+class HIIOSMRasterize(HIITask):
+    """
+
+    Process:
+
+    1. Fetch OSM pbf file
+    2. Convert PBF file -> Text file (filter by attribute/tag list)
+    3.
+        a)Split up text file by attribute and tags and 1 million row CSV files
+        b)Clean geometry and write out road tags to a roads CSV file
+    4. Rasterize each CSV file.
+    5. Merge all tiff into 1 multiband tiff file
+    6. Upload to Google Storage
+    7. Clean up working directories and files.
+
+    """
+
+    ee_osm_root = "osm"
+    google_creds_path = "/.google_creds"
+    _asset_prefix = f"projects/{HIITask.ee_project}/{ee_osm_root}"
+
+    MIN_GEOM_AREA = 5  # in meters
+    POLYGON_PRECISION = 5
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(self, *args, **kwargs)
+
+        self._args = kwargs
+        self.max_rows = 1000000
+
+        _extent = self._args.get("extent")
+        if _extent:
```

Comment:
Or maybe this test isn't behaving as expected? i.e. `argparse` is returning something truthy?